pkg/reconciler/nscert/nscert.go

```diff
@@ -143,6 +145,8 @@ func (c *reconciler) reconcile(ctx context.Context, ns *corev1.Namespace) error
 	} else if !equality.Semantic.DeepEqual(existingCert.Spec, desiredCert.Spec) {
```

Comment:
Should we check labels as well here, since they may be changed independently?

Reply:
Sounds reasonable to me.